homeassistant/components/blink/binary_sensor.py

```diff
@@ -45,39 +47,50 @@ async def async_setup_entry(
     hass: HomeAssistant, config: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up the blink binary sensors."""
-    data = hass.data[DOMAIN][config.entry_id]
+    coordinator: BlinkUpdateCoordinator = hass.data[DOMAIN][config.entry_id]
 
     entities = [
-        BlinkBinarySensor(data, camera, description)
-        for camera in data.cameras
+        BlinkBinarySensor(coordinator, camera, description)
+        for camera in coordinator.api.cameras
         for description in BINARY_SENSORS_TYPES
     ]
     async_add_entities(entities)
 
 
-class BlinkBinarySensor(BinarySensorEntity):
+class BlinkBinarySensor(CoordinatorEntity[BlinkUpdateCoordinator], BinarySensorEntity):
     """Representation of a Blink binary sensor."""
 
     _attr_has_entity_name = True
 
     def __init__(
-        self, data, camera, description: BinarySensorEntityDescription
+        self,
+        coordinator: BlinkUpdateCoordinator,
+        camera,
+        description: BinarySensorEntityDescription,
     ) -> None:
         """Initialize the sensor."""
-        self.data = data
+        super().__init__(coordinator)
         self.entity_description = description
-        self._camera = data.cameras[camera]
+        self._attr_name = f"{DOMAIN} {camera} {description.name}"
```

Comment:
Where does this name come from? I think this still a merge leftover :)

Reply:
Funny I didn't notice that IRL - 
blink Front door blink Front door UndefinedType._singleton

Reply:
hehe